resources/js/components/input/BorderedInput.jsx

```diff
@@ -22,17 +22,32 @@ const BorderedInput = ( { type, label, labelHidden, instructions, name, id, clas
 		className,
 	);
 
-	return (
-		<div>
-			{ label && <Label htmlFor={ id || name } hidden={ labelHidden } instructions={ instructions }>{ label }</Label> }
-			<input
-				id={ id || name }
-				name={ name }
-				type={ type }
-				className={ classes }
-				{ ...props }
-			/>
-		</div>
+	const formElementLabel = React.createElement(
+		Label,
+		{
+			htmlFor: id || name,
+			hidden: labelHidden,
+			instructions,
+		},
+		label
+	);
+
+	const formElement = React.createElement(
+		'textarea' === type ? 'textarea' : 'input',
+		{
+			id: ( id || name ),
+			name,
+			type,
+			className: classes,
+			...props,
+		}
+	);
+
+	return React.createElement(
+		'div',
+		{},
+		formElementLabel,
+		formElement
 	);
```

Comment:
I was thinking of just using `React.createElement` for the element that needs to have a dynamic type and still using JSX for the rest.

```suggestion
	return (
		<div>
			<Label 
				htmlFor={ id || name } 
				hidden={ labelHidden } 
				instructions={ instructions }
			>
				{label}
			</Label>
			{React.createElement(
				type === 'textarea' ? type : 'input',
				{
					id: id || name,
					type: type === 'textarea' ? null : type,
					name,
					className: classes,
					...props,
				},
			)}
		</div>
	);
```